distribution/systemd/vector.service

```diff
@@ -5,13 +5,44 @@ After=network-online.target
 Requires=network-online.target
 
 [Service]
+EnvironmentFile=-/etc/default/vector
 User=vector
 Group=vector
 ExecStart=/usr/bin/vector
 ExecReload=/bin/kill -HUP $MAINPID
 Restart=no
+
+# capabilities
 AmbientCapabilities=CAP_NET_BIND_SERVICE
-EnvironmentFile=-/etc/default/vector
+CapabilityBoundingSet=CAP_NET_BIND_SERVICE
+
+# sandboxing
+ProtectHostname=yes
+ProtectClock=yes
+ProtectKernelTunables=yes
+ProtectKernelModules=yes
+ProtectKernelLogs=yes
+ProtectSystem=strict
+ProtectHome=yes
+StateDirectory=vector
+ProtectControlGroups=yes
+PrivateTmp=yes
+PrivateDevices=yes
+RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
+LockPersonality=yes
+MemoryDenyWriteExecute=yes
+RestrictRealtime=yes
+RestrictSUIDSGID=yes
+NoNewPrivileges=yes
+RemoveIPC=yes
+RestrictNamespaces=yes
+
+# syscall filtering
+SystemCallFilter=@aio @basic-io @debug @file-system @io-event @ipc @network-io @signal @sync @timer
```

Comment:
It seems to me that:

* `@aio` we aren't using yet, but will probably want to take advantage of this at some point
* `@basic-io` is needed for reading/writing files
* `@debug` will be useful in the future for more advanced process instrumentation
* `@file-system` needed to create files and read directories
* `@io-event` I'm assuming tokio uses this for epoll and friends
* `@ipc` seems to be required for reading pipes and, in the future, I could see us supporting reading from POSIX queues
* `@network-io` needed for reading/writing to network
* `@signal` needed for signal handling
* `@sync`needed for fsync
* `@timer` I'm unsure if we actually use these

https://www.freedesktop.org/software/systemd/man/systemd.exec.html#System%20Call%20Filtering

Reply:
Pretty sure you should just use: `@debug @system-service`. `@system-service` includes all the other ones you've included, plus more. I know more isn't better in this case, but you're definitely missing some syscalls with your list (e.g., readdir).

Reply:
Ah, yes, you are right, `readdir` is not covered in `@file-system` like I expected it to be.

`@system-service` does seem like a reasonable base with the addition of `@debug`. I'll make that change.

I'll also plan to run this with some different configurations to make sure we aren't missing any required syscalls.

Thanks for taking a look!